isisd/isis_circuit.c

```diff
@@ -188,6 +188,10 @@ struct isis_circuit *isis_circuit_new(struct interface *ifp, const char *tag)
 	if (ifp->ifindex != IFINDEX_INTERNAL)
 		isis_circuit_enable(circuit);
 
+	circuit->ip_addrs = list_new();
+	circuit->ipv6_link = list_new();
+	circuit->ipv6_non_link = list_new();
+
```

Comment:
As you moved lists creation from `if_add` to `circuit_new`, you should also move lists deletion from `if_del` to `circuit_del`. Otherwise there will be a crash after `if_del` + `if_add`.

Reply:
yep you are correct, moving